### ISSUE TITLE
Building serialized database from a directory doesn't work 

### ISSUE THREAD

**Issue by ericolo** [jolespin/pyhmmsearch#1]
Hi ! 

I was trying to build a serialized database from TIGRFAM HMMs but it didn't work when doing it from the directory directly, like this:

`serialize_hmm_models.py -d hmm_PGAP/ -b TIGRFAM.pkl.gz`

It created the database without any error message but it was empty

I checked that the HMMs are indeed in that directory which was downloaded and unarchived from: https://ftp.ncbi.nlm.nih.gov/hmm/15.0/hmm_PGAP.HMM.tgz 

I hope i didn't miss something in the tutorial/manual this time ! 

(To go around this, I concatenated all HMM `cat hmm_PGAP/*.HMM > TIGRFAM.HMM` and ran `serialize_hmm_models.py -d TIGRFAM.HMM -b TIGRFAM.pkl.gz`, and that worked)

**Comment by jolespin** [comment]
I think you figured out the error but the PyHMMSearch tool expects a concatenated database  (similar to Pfam) and NOT a directory of individual HMMs (as PyKOfamSearch expects).  The workaround you have is the intended usage for databases that aren't concatenated. 

Looking at the docs now and it's not very clear that this is the intended usage so I'll add that in there and will fail if there is a directory: 
```
Database arguments:
  -d HMM_DATABASE, --hmm_database HMM_DATABASE
                        path/to/database.hmm cannot be used with -b/-serialized_database
```

**Comment by jolespin** [comment]
Figured it out. When it was checking for extensions, it was globbing with "." characters. 

    if opts.hmm_database:
        assert opts.hmm_list == "stdin", "Cannot provide --hmm_database and --hmm_list"
        assert stdin_is_empty(), "Cannot provide stdin if using --hmm_database"
        if os.path.isdir(opts.hmm_database):
            for extension in [".hmm", ".hmm.gz", ".HMM", ".HMM.gz"]:
                for fp in glob.glob(os.path.join(opts.hmm_database, f"*.{extension}")):
                    hmm_filepaths.append(fp)

So it was actually looking for "TIGR0123..HMM" instead of "TIGR0123.HMM"

Thanks for catching this error. Fixing now. 

Needs to be this: 

```
    if opts.hmm_database:
        assert opts.hmm_list == "stdin", "Cannot provide --hmm_database and --hmm_list"
        assert stdin_is_empty(), "Cannot provide stdin if using --hmm_database"
        if os.path.isdir(opts.hmm_database):
            for extension in ["hmm", "hmm.gz", "HMM", "HMM.gz"]:
                for fp in glob.glob(os.path.join(opts.hmm_database, f"*.{extension}")):
                    hmm_filepaths.append(fp)
```

**Comment by jolespin** [comment]
Fixed in v2024.7.29: https://pypi.org/project/pyhmmsearch/2024.7.29/

**Comment by ericolo** [comment]
That's great ! Thanks for the quick reply and quick fix !
